components/event-publisher/org.wso2.carbon.event.publisher.core/src/main/java/org/wso2/carbon/event/publisher/core/internal/EventPublisher.java

```diff
@@ -145,6 +147,33 @@ public void sendEvent(Event event) {
         if (isPolled || !EventPublisherServiceValueHolder.getCarbonEventPublisherManagementService().isDrop()) {
             process(event);
         }
+
+        if(EventPublisherServiceValueHolder.getEventManagementService().getManagementModeInfo().getMode() == Mode.HA){
+            if(!isPolled && EventPublisherServiceValueHolder.getCarbonEventPublisherManagementService().isDrop()){
+                //add to Queue
+                EventWrapper eventWrapper = new EventWrapper(event, EventPublisherServiceValueHolder.getEventManagementService().getClusterTimeInMilies());
+                eventQueue.add(eventWrapper);
+
+                // get last processed time and remove old events from the queue
+                long timestamp = EventPublisherServiceValueHolder.getEventManagementService().getLatestEventSentTime(
+                        eventPublisherConfiguration.getEventPublisherName(), tenantId);
+
+                while (!eventQueue.isEmpty() && eventQueue.peek().getTimestampInMilies() <= timestamp) {
+                    eventQueue.remove();
+                }
+
+            }else if(!isPolled && !EventPublisherServiceValueHolder.getCarbonEventPublisherManagementService().isDrop()){
```

Comment:
Thanks @nirmal070125, The changes are done in the pull request - https://github.com/wso2/carbon-analytics-common/pull/121/files